contracts/aggregator/polygon/flashloan/main.sol

```diff
@@ -184,6 +251,37 @@ contract FlashAggregatorPolygon is Helper {
         balancerLending.flashLoan(InstaFlashReceiverInterface(address(this)), wethTokenList_, wethAmountList_, data_);
     }
 
+    /**
+     * @dev Middle function for route 9.
+     * @notice Middle function for route 9.
+     * @param _tokens list of token addresses for flashloan.
+     * @param _amounts list of amounts for the corresponding assets.
+     * @param _data extra data passed.
+    */
+    function routeInterop(address[] memory _tokens, uint256[] memory _amounts, bytes memory _data) internal {
+        bytes memory data_ = abi.encode(8, _tokens, _amounts, msg.sender, _data);
+        dataHash = bytes32(keccak256(data_));
+        interopLending.initiateFlashLoan(address(this), _tokens, _amounts, data_);
+    }
+
+    /**
+     * @dev Middle function for route 9.
+     * @notice Middle function for route 9.
+     * @param _tokens list of token addresses for flashloan.
+     * @param _amounts list of amounts for the corresponding assets.
+     * @param _data extra data passed.
+     * @param _instaData extra data passed.
+    */
+    function routeInteropAave(address[] memory _tokens, uint256[] memory _amounts, bytes memory _data, bytes memory _instaData) internal {
+        (address[] memory tokens_, uint256[] memory amounts_) = abi.decode(
+            _instaData,
+            (address[], uint256[])
+        );
+        bytes memory data_ = abi.encode(9, _tokens, _amounts, msg.sender, _data);
+        dataHash = bytes32(keccak256(data_));
+        interopLending.initiateFlashLoan(address(this), tokens_, amounts_, data_);
```

Comment:
@Vaibhav20201 can you rename the `tokens` and `amounts` variables according to there usage? Getting bit confused

Reply:
Sure

Reply:
done